src/compiler/types.ts

```diff
@@ -8685,29 +8685,74 @@ export interface TransformationResult<T extends Node> {
  * A function that is used to initialize and return a `Transformer` callback, which in turn
  * will be used to transform one or more nodes.
  */
-export type TransformerFactory<T extends Node> = (context: TransformationContext) => Transformer<T>;
+export type TransformerFactory<TIn extends Node, TOut extends Node | undefined = TIn> = (context: TransformationContext) => Transformer<TIn, TOut>;

```

Comment:
What transformer factories allow you to return `undefined`? That shouldn't be permitted.

Reply:
This is just the type variable propagated up from `Transformer`. Changing it to be `TOut extends Node = TIn` doesn't cause any compiler errors in our own codebase. (The same applies to Transformer, too.)